flexget/plugins/output/send_telegram.py

```diff
@@ -0,0 +1,347 @@
+            item = cached_usernames.get(username)
+            if item:
+                chat_ids.append(item)
+                usernames.pop(len_ - i - 1)
+
+        len_ = len(fullnames)
+        for i, fullname in enumerate(reversed(fullnames)):
+            item = cached_fullnames.get(fullname)
+            if item:
+                chat_ids.append(item)
+                fullnames.pop(len_ - i - 1)
+
+        len_ = len(groups)
+        for i, grp in enumerate(reversed(groups)):
+            item = cached_groups.get(grp)
+            if item:
+                chat_ids.append(item)
+                groups.pop(len_ - i - 1)
+
+        return chat_ids
+
+    def _get_new_chat_ids(self, bot, usernames, fullnames, groups):
+        """get chat ids by querying the telegram `bot`
+
+        :type bot: telegram.Bot
+        :type usernames: list[str]
+        :type fullnames: list[tuple[str, str]]
+        :type groups: list[str]
+        :rtype: __generator[ChatIdEntry]
+
+        """
+        upd_usernames, upd_fullnames, upd_groups = self._get_bot_updates(bot)
+
+        len_ = len(usernames)
+        for i, username in enumerate(reversed(usernames)):
+            chat = upd_usernames.get(username)
+            if chat is not None:
+                entry = ChatIdEntry(id=chat.id, username=chat.username, firstname=chat.first_name,
+                                    surname=chat.last_name)
+                yield entry
+                usernames.pop(len_ - i - 1)
+
+        len_ = len(fullnames)
+        for i, fullname in enumerate(reversed(fullnames)):
+            chat = upd_fullnames.get(fullname)
+            if chat is not None:
+                entry = ChatIdEntry(id=chat.id, username=chat.username, firstname=chat.first_name,
+                                    surname=chat.last_name)
+                yield entry
+                fullnames.pop(len_ - i - 1)
+
+        len_ = len(groups)
+        for i, grp in enumerate(reversed(groups)):
+            chat = upd_groups.get(grp)
+            if chat is not None:
+                entry = ChatIdEntry(id=chat.id, group=chat.title)
+                yield entry
+                groups.pop(len_ - i - 1)
+
+    def _get_bot_updates(self, bot):
+        """get updated chats info from telegram
+        :type bot: telegram.Bot
+        :rtype: (dict[str, telegram.User], dict[(str, str), telegram.User], dict[str, telegram.GroupChat])
+
+        """
+        updates = bot.getUpdates()
+        usernames = dict()
+        fullnames = dict()
+        groups = dict()
+
+        for chat in (x.message.chat for x in updates):
+            if isinstance(chat, telegram.User):
+                usernames[chat.username] = chat
+                fullnames[(chat.first_name, chat.last_name)] = chat
+            elif isinstance(chat, telegram.GroupChat):
+                groups[chat.title] = chat
+            else:
+                self.log.warn('unknown chat type: {}'.format(type(chat)))
```

Comment:
Personally I think `.format` is ugly, but it's compatible with Python 2.6, so I guess it's ok. It also seems to be the standard in future Python 3 versions.